pkg/azurefile/azurefile.go

```diff
@@ -864,6 +899,66 @@ func (d *Driver) ResizeFileShare(ctx context.Context, subsID, resourceGroup, acc
 	})
 }
 
+// CopyFileShare copies a fileshare in the same storage account
+func (d *Driver) CopyFileShare(ctx context.Context, req *csi.CreateVolumeRequest, accountKey string, shareOptions *fileclient.ShareOptions, storageEndpointSuffix string, secrets map[string]string) error {
+	if shareOptions.Protocol == storage.EnabledProtocolsNFS {
+		return fmt.Errorf("protocol nfs is not supported for volume cloning")
+	}
+	var volumeID string
+	if req.GetVolumeContentSource() != nil {
+		if req.GetVolumeContentSource().GetVolume() != nil {
+			volumeID = req.GetVolumeContentSource().GetVolume().GetVolumeId()
+		}
+	}
+	resourceGroupName, accountName, srcFileShareName, _, _, subsID, err := GetFileShareInfo(volumeID)
+	dstFileShareName := shareOptions.Name
+	if err != nil {
+		return status.Error(codes.NotFound, err.Error())
+	}
+	klog.V(2).Infof("begin to copy fileshare %s to %s", srcFileShareName, dstFileShareName)
+	if srcFileShareName == "" {
+		return fmt.Errorf("srcFileShareName is empty")
+	}
+	if dstFileShareName == "" {
+		return fmt.Errorf("dstFileShareName is empty")
+	}
+	if len(secrets) > 0 {
+		accountSasToken, err := getAccountSasToken(secrets, accountName)
+		if err != nil || accountSasToken == "" {
+			klog.V(2).Infof("get account sas token from secrets err: %s, generate sas token for account(%s)", err.Error(), accountName)
+			accountSasToken, err = generateSASToken(accountName, accountKey, storageEndpointSuffix)
+			if err != nil {
+				return err
+			}
+		}
+		exists, err := d.fileClient.existFileShare(accountName, accountKey, srcFileShareName)
+		if err != nil {
+			return err
+		}
+		if !exists {
+			return fmt.Errorf("source fileshare(%s) does not exist", srcFileShareName)
+		}
+
+		if err := d.waitForAzCopySuccess(ctx, accountName, resourceGroupName, storageEndpointSuffix, srcFileShareName, dstFileShareName, accountSasToken, waitForCopyInterval, waitForCopyTimeout); err != nil {
+			return fmt.Errorf("failed to wait for azcopy success: %v", err.Error())
+		}
+	} else {
+		klog.V(2).Infof("generate sas token for account(%s)", accountName)
+		accountSasToken, genErr := generateSASToken(accountName, accountKey, storageEndpointSuffix)
+		if genErr != nil {
+			return genErr
+		}
+		if _, rerr := d.cloud.GetFileShare(ctx, subsID, resourceGroupName, accountName, srcFileShareName); rerr != nil {
+			return fmt.Errorf("srcFileShareName(%s) get error: %s", srcFileShareName, rerr.Error())
+		}
+
+		if err := d.waitForAzCopySuccess(ctx, accountName, resourceGroupName, storageEndpointSuffix, srcFileShareName, dstFileShareName, accountSasToken, waitForCopyInterval, waitForCopyTimeout); err != nil {
```

Comment:
don't use duplicated code as L942 are also calling `d.waitForAzCopySuccess`

Reply:
done, remove this func